### PR TITLE
Use pytest as testrunner for unit tests #359

### PR DESCRIPTION
**Pull request by Motwani**
PR fixes issue #359

@whimboo Please Review

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -9,6 +9,10 @@ deps =
     mozprocess >= 0.22
     moztest >= 0.7
     wptserve >= 1.4.0
+    pytest
+    coverage
+    pytest-cov
 
 commands =
-    {toxinidir}/tests/test.py
```
Comment (whimboo) [inline tox.ini]: Does it mean we can remove test.py now?

File: tox.ini [hunk 2]
```diff
@@ -9,6 +9,10 @@ deps =
     mozprocess >= 0.22
     moztest >= 0.7
     wptserve >= 1.4.0
+    pytest
+    coverage
+    pytest-cov
```
Comment (whimboo) [inline tox.ini]: Please sort them all alphabetically.

File: tox.ini [hunk 3]
```diff
@@ -9,6 +9,10 @@ deps =
...
 
 commands =
-    {toxinidir}/tests/test.py
+        py.test --cov=mozdownload
+        coverage report
```
Comment (whimboo) [inline tox.ini]: I would suggest to remove coverage for now given that it still does not work. Also the second line here would result in a duplicated output of the report.

**Comment by whimboo** [comment]
Overall this PR looks great! I tested it locally and I really like the pytest solution. I made some comments which I would like to see fixed before I can merge the PR. Thanks!

File: tox.ini (hunk 1 above)
Comment (Motwani) [inline tox.ini]: Yes all 48 tests are collected and run without test.py

File: tox.ini [hunk 4]
```diff
@@ -11,4 +12,4 @@ deps =
     wptserve >= 1.4.0
```
Comment (whimboo) [inline tox.ini]: manifestparser and moztest are no longer used now given that `test.py` got deleted. So please also kill those dependencies.

**Comment by Motwani** [comment]
@whimboo Updated PR , Please review

File: tox.ini [hunk 5]
```diff
@@ -3,12 +3,11 @@ envlist = py27
 
 [testenv:py27]
 deps =
-    manifestparser >= 1.1
+    pytest
```
Comment (whimboo) [inline tox.ini]: Put that in alphabetical order and we are fine. Sorry for not noticing that earlier.

File: tox.ini [hunk 6]
```diff
@@ -3,12 +3,11 @@ envlist = py27
...
     mock
     mozfile >= 1.1
     mozprocess >= 0.22
-    moztest >= 0.7
     wptserve >= 1.4.0
 
 commands =
-    {toxinidir}/tests/test.py
+        py.test
```
Comment (whimboo) [inline tox.ini]: Why those extra blanks? 4 was totally fine.

**Comment by whimboo** [comment]
pytest is still not in the right order but I will fix it before squashing all commits and merging this PR. Thanks Aditya for the work!

**Comment by whimboo** [comment]
While running locally, I also noticed that pytest creates a `.cache` folder in the root of the repository checkout. I will add this to the `.gitignore` file too.

**Comment by whimboo** [comment]
Merged PR to master as https://github.com/mozilla/mozdownload/commit/2be59688fcdebddf22eddc569f6090030b486269.

**Pull request closed without merging**